pkg/sql/opt/norm/testdata/rules/join

```diff
@@ -321,6 +321,390 @@ anti-join
+ │                   └── group-by
+ │                        ├── columns: min:10(int)
+ │                        ├── cardinality: [1 - 1]
+ │                        ├── key: ()
+ │                        ├── fd: ()-->(10)
+ │                        ├── scan b
+ │                        │    ├── columns: b.x:8(int!null)
+ │                        │    └── key: (8)
+ │                        └── aggregations [outer=(8)]
+ │                             └── min [type=int, outer=(8)]
+ │                                  └── variable: b.x [type=int, outer=(8)]
+ ├── scan b
+ │    ├── columns: b.x:6(int!null) b.y:7(int)
+ │    ├── key: (6)
+ │    └── fd: (6)-->(7)
+ └── filters [type=bool, outer=(1,6), constraints=(/1: (/NULL - ]; /6: (/NULL - ]), fd=(1)==(6), (6)==(1)]
+      └── a.k = b.x [type=bool, outer=(1,6), constraints=(/1: (/NULL - ]; /6: (/NULL - ])]
+
+# Optimization does not apply with correlated suqueries.
+opt
+SELECT * FROM a JOIN b ON a.k = b.x AND b.x * a.i = (SELECT min(a.k) FROM b)
+----
+project
+ ├── columns: k:1(int!null) i:2(int) f:3(float) s:4(string) j:5(jsonb) x:6(int!null) y:7(int)
+ ├── key: (1,6)
+ ├── fd: (1)-->(2-5), (6)-->(7)
+ └── inner-join-apply
+      ├── columns: a.k:1(int!null) i:2(int) f:3(float) s:4(string) j:5(jsonb) b.x:6(int!null) b.y:7(int) min:11(int)
+      ├── key: (1,6)
+      ├── fd: (1)-->(2-5), (6)-->(7,11)
+      ├── scan a
+      │    ├── columns: a.k:1(int!null) i:2(int) f:3(float) s:4(string) j:5(jsonb)
+      │    ├── key: (1)
+      │    └── fd: (1)-->(2-5)
+      ├── group-by
+      │    ├── columns: b.x:6(int!null) b.y:7(int) min:11(int)
+      │    ├── grouping columns: b.x:6(int!null)
+      │    ├── outer: (1)
+      │    ├── key: (6)
+      │    ├── fd: (6)-->(7,11)
+      │    ├── left-join
+      │    │    ├── columns: b.x:6(int!null) b.y:7(int) k:10(int)
+      │    │    ├── outer: (1)
+      │    │    ├── fd: (6)-->(7), ()~~>(10)
+      │    │    ├── scan b
+      │    │    │    ├── columns: b.x:6(int!null) b.y:7(int)
+      │    │    │    ├── key: (6)
+      │    │    │    └── fd: (6)-->(7)
+      │    │    ├── project
+      │    │    │    ├── columns: k:10(int)
+      │    │    │    ├── outer: (1)
+      │    │    │    ├── fd: ()-->(10)
+      │    │    │    ├── scan b
+      │    │    │    └── projections [outer=(1)]
+      │    │    │         └── variable: a.k [type=int, outer=(1)]
+      │    │    └── true [type=bool]
+      │    └── aggregations [outer=(7,10)]
+      │         ├── min [type=int, outer=(10)]
+      │         │    └── variable: k [type=int, outer=(10)]
+      │         └── any-not-null [type=int, outer=(7)]
+      │              └── variable: b.y [type=int, outer=(7)]
+      └── filters [type=bool, outer=(1,2,6,11), constraints=(/1: (/NULL - ]; /6: (/NULL - ]), fd=(1)==(6), (6)==(1)]
+           ├── a.k = b.x [type=bool, outer=(1,6), constraints=(/1: (/NULL - ]; /6: (/NULL - ])]
+           └── (b.x * a.i) = min [type=bool, outer=(2,6,11)]
+
```

Comment:
How about a test case that involves equality conditions that only involve columns from one side or the other? The idea is to try and confuse the rules with a non-join equality.